### PR TITLE
Adding methods from parent classes in to children classes

### PR DESCRIPTION
**Pull request by SenyaMur**
Implementation of adding methods from parent classes to children for all entity sets and singletons.
See description in #46

### CODE REVIEW

**Comment by xuzhg** [comment]
@SenyaMur Cool, Thanks for your PR. I will review it later.

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Operation/EdmFunctionOperationHandlerTests.cs [hunk 1]
```diff
@@ -51,6 +51,41 @@ public void CreateOperationForEdmFunctionReturnsCorrectOperation()
             Assert.Equal(2, operation.Responses.Count);
             Assert.Equal(new string[] { "200", "default" }, operation.Responses.Select(e => e.Key));
         }
+        [Fact]
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Operation/EdmFunctionOperationHandlerTests.cs]: add a new white line

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiPathsGeneratorTests.cs [hunk 2]
```diff
@@ -66,5 +66,30 @@ public void CreatePathsReturnsForBasicModel()
...
+            var paths = context.CreatePaths();
+
+            // Assert
+            Assert.NotNull(paths);
+            Assert.Equal(4, paths.Count);
+
+            Assert.Contains("/Accounts", paths.Keys);
+            Assert.Contains("/Accounts/{id}", paths.Keys);
+            Assert.Contains("/Accounts/{id}/Attachments()", paths.Keys);
+            Assert.Contains("/Accounts/{id}/AttachmentsAdd", paths.Keys);
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Generator/OpenApiPathsGeneratorTests.cs]: >             [](start = 73, length = 12)

remove the extra whitespace

File: test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/Contract.OData.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      <Action Name="AttachmentsAdd" IsBound="true">
+        <Parameter Name="bindingParameter" Type="Microsoft.OData.Service.Sample.Contract.BaseEntityApiModel"/>
+        <Parameter Name="comment" Type="Edm.String"/>
+        <Parameter Name="file" Type="Edm.String"/>
+      </Action>
+      <EntityContainer Name="Container">        
+        <EntitySet Name="Accounts" EntityType="Microsoft.OData.Service.Sample.Contract.AccountApiModel">          
+        </EntitySet>
+      </EntityContainer>
+		</Schema>
```
Comment (xuzhg) [inline test/Microsoft.OpenAPI.OData.Reader.Tests/Resources/Contract.OData.xml]: can we make good alignment

**Comment by xuzhg** [comment]
@SenyaMur the code looks good to me. I will merge and change some small comments.

---
In reply to: [573122512](https://github.com/microsoft/OpenAPI.NET.OData/pull/47#issuecomment-573122512) [](ancestors = 573122512)

**Comment by SenyaMur** [comment]
@xuzhg  thank's for your approve

**Pull request merged**